### PR TITLE
DO NOT Merge - WIP Add ctx span to index view, add lccn/oclcnum to ctx openurl.

### PR DESCRIPTION
**Pull request by kevinreiss**

### CODE REVIEW

**Comment by jpstroop** [comment]
I think this could use a couple of specs

**Comment by kevinreiss** [comment]
I added a few basis feature specs since it is still unclear how much of this functionality will move to the Orangeindex/Traject process (for OpenURL generation) and a standalone/embedded engine for library requests (Copies in Library).

**Comment by tampakis** [comment]
I've been going back and forth on this. The important bit is that the URL contains the params/values that Traject extracts. I'm just not sure if the URL should be compiled in Orangeindex or Orangelight.

**Comment by kevinreiss** [comment]
Okay. The current method is pretty heavy to execute to generate the <span> tag Zotero can scrape off the results. I will re-write the template to not parse the marc and just generate all values from the normalized data in the solr doc. How does that sound? 

I can also address #218 with a further revision of this.

File: config/initializers/blacklight_initializer.rb [hunk 1]
```diff
@@ -65,7 +66,14 @@ def export_as_openurl_ctx_kev(format = nil)
         export_text << "&amp;rft.isbn=#{(isbn.nil? or isbn['a'].nil?) ? "" : isbn['a']}"
       end
     end
-    export_text << "&amp;rft_id=" + (id.nil? ? "" : CGI::escape("pul_"+id.value))
+    
+    export_text << "&amp;rft_id=" + (id.nil? ? "" : CGI::escape("http://bibdata.princeton.edu/bibliographic/"+id.value))
```
Comment (tampakis) [inline config/initializers/blacklight_initializer.rb]: How does the rft_id value get used?
Stylewise I think it's more readable to interpolate the string: `"http://bibdata.princeton.edu/bibliographic/{id.value}"`

Comment (kevinreiss) [inline config/initializers/blacklight_initializer.rb]: rft_id is catchall parameter in OpenURL that can be repeated that is supposed to contain either a reference to an identifier like a DOI, LCCN, or OCLCnum or just a basic encoded URL for the resource in question. It is up to the OpenURL resolver to do something with them.

**Comment by kevinreiss** [comment]
Still working on this per earlier comment from @tampakis about how to construct the OpenURL. Don't merge.

**Comment by kevinreiss** [comment]
Hey, I tried working through generating a good citation from the solr index and in order to do that effectively we are going to have make some granular field definitions for things related to dates and publications. The MARC data has a richer set of data than we have normalized into the index. We should probably do that in the context of a different branch than this group of changes.

File: config/requests.yml.tmpl [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# requests.yml
+
+defaults: &defaults
+  umlaut_base: https://getit.princeton.edu
+  bibdata_base: http://bibdata.princeton.edu/bibliographic
```
Comment (jpstroop) [inline config/requests.yml.tmpl]: Should any code that uses `http://bibdata.princeton.edu/locations` be refactored to use config/constants from this file? Or is that all Javascript? js.erb? or is that bad practice?

Comment (kevinreiss) [inline config/requests.yml.tmpl]: That is what I was thinking so we have a clean way to point at test versions of external services like bibdata down the road. I think we can  expose config values to the javascript layer directly.

Comment (kevinreiss) [inline config/requests.yml.tmpl]: If we wind up adding a lot of config that needs to be exposed to JS we could create some sort of config object that gets registered in the Blacklight JS namespace. For a few values we could just use data attributes that our custom JS knows to pull when it needs something like this.

**Pull request closed without merging**